wgpu/gui/auto.py

```diff
@@ -27,9 +28,35 @@ def is_jupyter():
 else:
     try:
         from .glfw import WgpuCanvas, run, call_later  # noqa
-    except ImportError as err:
-        msg = str(err)
-        msg += "\n\n  Install glfw using e.g. ``pip install -U glfw``."
-        if sys.platform.startswith("linux"):
-            msg += "\n  You may also need to run the equivalent of ``apt install libglfw3``."
-        raise ImportError(msg) from None
+    except ImportError as glfw_err:
+        try:
+            for libname in ("PySide6", "PyQt6", "PySide2", "PyQt5", "PySide", "PyQt4"):
+                try:
+                    importlib.import_module(libname)
+                    break
+                except ImportError:
+                    pass
+
+            from .qt import WgpuCanvas, QtWidgets, QtCore
+
+            # When using Qt, there needs to be an
+            # application before any widget is created
+            app = QtWidgets.QApplication([])
+
+            def run():
+                app.exec() if hasattr(app, "exec") else app.exec_()
+
+            def call_later(delay, callback, *args):
+                QtCore.QTimer.singleShot(delay * 1000, lambda: callback(*args))
```

Comment:
I hope we can move `run()` and `call_later` in `qt.py`, so that people can do `from wgpu.gui.qt import WgpuCanvas, run`, e.g. to force using qt if glfw is available ;)

I think we need to do something smart with the instantiation of the application. Maybe, in the init of Qt's WGpuCanvas we can check whether an application exists, and we create it if not? Maybe there is a more elegant way?

BTW: Glfw has a similar issue, but people using glfw are not going to embed it into their glfw GUI app, so we can take control and initialize glfw.

Reply:
Hmm, I see what you mean, but a check during instantiating seems a bit dirty.
What I like about using the `auto` module for instantiating the app is that there isn't too much cruft leaking into the `qt` module.
We could do something like: `from wgpu.gui.auto.qt import WgpuCanvas, run` for users that specifically would want to use Qt. That should make it explicit that the user wants to use the auto module with the qt framework and leave the `wgpu.gui.qt` module as is.

Reply:
I've gone ahead and separated the auto gui functionality from the `wgpu.gui.qt` module into the `wgpu.gui.auto.qt` module. Hope you like it! We could also apply the same for glfw of course, but maybe in another PR, together with the de-tri-plication?

Reply:
Ok, reverted the creation of `wgpu.gui.auto.qt`!